### PR TITLE
Good default build options

### PR DESCRIPTION
**Pull request by whit2333**
Set configure variables to (subjectively) good defaults.
    
Here are the defaults and my reasoning.
    
 - `DD4HEP_USE_XERCESC` is `ON`, otherwise what is the point, right?
 - `DD4HEP_USE_PYROOT`  is `ON`, again lots of usefulness is thrown away otherwise.
 - `DD4HEP_USE_GEANT4`  is `ON`. Are there any use cases *without Geant4*?
 - `DD4HEP_USE_GEAR`    is `OFF`. This is a legacy issue. 
 - `DD4HEP_USE_LCIO`    is `OFF`. IMO the only practical option.
 - `BUILD_TESTING`      is `ON`. Has anybody tried building with it set to `OFF`?
 - `DD4HEP_USE_CXX14`   is `ON`, because I am an optimist living in 2017 :)
    
    Here are the arguments for these changes:
    
    1. The minimum number of cmake arguments should configure a build that  will compile. That is the following will just work.
    ```
    cmake source/. -DCMAKE_INSTALL_PREFIX=$HOME
    make -j4 install
    ```
    
    2. Most of the options if turned off (which they are by default) defeat
    the purpose of using DD4hep. That is GEANT4 and pyroot should not be
    optional dependencies.
    
    3. If, for example, GEANT4 is not required for the build or not found with the
    `find_package` call, then `DDHEP_USE_GEANT4` can be turned off.
    
    Here is an implementation idea
    ```
    set( IS_GEANT4_REQUIRED )
    if(DD4HEP_USE_GEANT4 STREQUAL "ON")
      set( IS_GEANT4_REQUIRED "REQUIRED")
    endif(DD4HEP_USE_GEANT4 STREQUAL "ON")
    if(NOT DD4HEP_USE_GEANT4 STREQUAL "OFF")
      find_package( Geant4 9.6 ${IS_GEANT4_REQUIRED} gdml ui_all vis_all)
    endif(NOT DD4HEP_USE_GEANT4 STREQUAL "OFF")
    ```
    If `DD4HEP_USE_GEANT4` is not set to any value, GEANT4 is used if it is
    found.

BEGINRELEASENOTES
- Changed the default build options. They now are: 
   - `DD4HEP_USE_XERCESC` is  `ON`
   - `DD4HEP_USE_PYROOT`  is `ON`
   - `DD4HEP_USE_GEANT4` is `ON`

ENDRELEASENOTES

### CODE REVIEW

**Comment by MarkusFrankATcernch** [comment]
Hi Whit,

1) DD4HEP_USE_PYROOT should be OFF.
Actually it should be ditched completely.
This has nothing to do with using DD4hep with python.
This is pure legacy, unmaintained and misleading.
If one really would want to write detector constructors with python
the implementation should be re-thought.

2) XercesC **is** optional  => OFF
For nearly all practical purposes TinyXML is sufficient.

Rest sounds reasonable. Like the optimist part ;-)


Markus
________________________________

**Comment by andresailer** [comment]
Please drop the commit that is also in #171 https://github.com/AIDASoft/DD4hep/pull/171/commits/296aa0edaabdf3f5ca541a32f3c7d8da20882e13

**Comment by whit2333** [comment]
@MarkusFrankATcernch, thanks for the info. I think I still had trouble without using XercesC but I did not investigate further.

File: cmake/DD4hepConfig.cmake.in [hunk 1]
```diff
@@ -25,9 +25,17 @@ set ( DD4HEP_USE_CXX14   "@DD4HEP_USE_CXX14@" )
...
+  message("Could not find ROOTSYS environment variable")
+  message("Using location at build time ROOT_INCLUDE_DIRS/.. @ROOT_INCLUDE_DIRS@/..")
+  set ( ROOTSYS            "@ROOT_INCLUDE_DIRS@/.." )
+  set ( ROOT_DIR           "@ROOT_INCLUDE_DIRS@/../cmake" )
+  set ( ROOT_VERSION       "@ROOT_VERSION@" )
+else()
+  set ( ROOTSYS            "$ENV{ROOTSYS}" )
+  set ( ROOT_DIR           "$ENV{ROOTSYS}/cmake" )
+  set ( ROOT_VERSION       "@ROOT_VERSION@" )
+endif(NOT ENV{ROOTSYS}))
```
Comment (petricm) [inline cmake/DD4hepConfig.cmake.in]: one `)` too much

**Comment by petricm** [comment]
The developers will not change the cmake defaults at this point. The `ROOTSYS` part is ok

**Comment by petricm** [comment]
The developers decided on the last meeting to retain current defaults

**Pull request closed without merging**